src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -262,8 +234,13 @@ void GcsResourceManager::HandleGetAllResourceUsage(
     reply->mutable_resource_usage_data()->CopyFrom(batch);
   }
 
-  RAY_DCHECK(static_cast<size_t>(reply->resource_usage_data().batch().size()) ==
-             num_alive_nodes_);
+  if (static_cast<size_t>(reply->resource_usage_data().batch().size()) ==
```

Comment:
is this intentional

Reply:
yeah, they might be different and we don't want to crash

Reply:
Comment on when they can be different

Reply:
When can they be different?